### DISCUSSION TITLE
What are the limitations of promitor?

### DISCUSSION THREAD

**Discussion by adhiraj-g** [tomkerkhove/promitor#1638] [Q&A] [answered]
Hi Tom,

would the Promitor agent scraper fetch out metrics associated to the AKS cluster only?

Would it be able to fetch out metrics of an azure infra setup that does not use containers?

What are the limitations of Promitor in that case? Also, Can Promitor be deployed outside azure infrastructure as well, provided there is connectivity between Azure and external system? Since, All the documentation, i could find is showing deployment on AKS only.

Looking forward to hear back from you soon.

**Comment by tomkerkhove** [comment, marked as answer]
Those are very good questions, here is my feedback:

> would the Promitor agent scraper fetch out metrics associated to the AKS cluster only?
> 
> Would it be able to fetch out metrics of an azure infra setup that does not use containers?

You can scrape practically *any* metric that is available in Azure Monitor emitted by *any* Azure service, not just the ones by AKS or related to containers.

> What are the limitations of Promitor in that case? Also, Can Promitor be deployed outside azure infrastructure as well, provided there is connectivity between Azure and external system? Since, All the documentation, i could find is showing deployment on AKS only.

The documentation has a walkthrough for AKS indeed, but you can run it on any container orchestrator that you want running anywhere. However, we provide Helm charts that simplify the deployment on Kubernetes which is the preferred orchestrator.

The only feature that you would not be able to use would be using pod identity as that relies on the Azure infrastructure, but you can still use service principle authentication for that. ([docs](https://promitor.io/configuration/v2.x/azure-authentication#supported-authentication-mechanisms))

Hope this helps?

**Comment by adhiraj-g** [comment]
Thank you so much for your feedback.
